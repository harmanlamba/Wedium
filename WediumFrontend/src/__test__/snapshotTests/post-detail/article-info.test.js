import React from 'react';
import { shallow } from 'enzyme';

import ArticleInfo from '../../../components/post-detail/article-info';

it('snapshot articleInfo default load', () => {
  const post = {
    articleTitle: 'Title',
    articleBody: 'Body',
    articleImageUrl: 'Image',
  };

  let wrapper = shallow(<ArticleInfo post={post} />);

  expect(wrapper).toMatchSnapshot();
});

it('snapshot postform load with missing image', () => {
  const MISSING_IMAGE_URL =
    'https://image.flaticon.com/icons/svg/570/570975.svg';

  const post = {
    articleTitle: 'Title',
    articleBody: '<p>Body</p>',
    articleImageUrl: MISSING_IMAGE_URL,
  };

  let wrapper = shallow(<ArticleInfo post={post} />);

  expect(wrapper).toMatchSnapshot();
});
