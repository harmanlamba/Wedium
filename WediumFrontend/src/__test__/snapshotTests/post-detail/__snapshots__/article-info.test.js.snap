// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot articleInfo default load 1`] = `
<div>
  <div
    className="makeStyles-imageCard-1"
  >
    <WithStyles(ForwardRef(CardMedia))
      className="makeStyles-image-2"
      component="img"
      src="Image"
    />
  </div>
  <div
    className="makeStyles-article-3"
  >
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-articleTitle-4"
      variant="h6"
    >
      Title
    </WithStyles(ForwardRef(Typography))>
    <t
      onExitHighlight={null}
      onHighlightComment={null}
      popoverItems={[Function]}
    >
      <div
        className="makeStyles-articleBody-5"
      >
        Body
      </div>
    </t>
  </div>
</div>
`;

exports[`snapshot postform load with missing image 1`] = `
<div>
  <div
    className="makeStyles-article-3"
  >
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-articleTitle-4"
      variant="h6"
    >
      Title
    </WithStyles(ForwardRef(Typography))>
    <t
      onExitHighlight={null}
      onHighlightComment={null}
      popoverItems={[Function]}
    >
      <div
        className="makeStyles-articleBody-5"
      >
        <p>
          Body
        </p>
      </div>
    </t>
  </div>
</div>
`;
